api/resolvers/wallet.js

```diff
@@ -456,6 +459,48 @@ const resolvers = {
         cursor: nextCursor,
         entries: logs
       }
+    },
+    failedInvoices: async (parent, args, { me, models }) => {
+      if (!me) {
+        throw new GqlAuthenticationError()
+      }
+      // make sure each invoice is only returned once via visibility timeouts and SKIP LOCKED
+      // we use queryRawUnsafe because Prisma does not support tsrange
+      // see https://www.prisma.io/docs/orm/overview/databases/postgresql
+      return await models.$queryRawUnsafe(`
+        WITH failed AS (
+          UPDATE "Invoice"
+          SET "lockedAt" = now()
+          WHERE id IN (
+            SELECT id FROM "Invoice"
+            WHERE "userId" = $1
+            AND "actionState" = 'FAILED'
+            AND "userCancel" = false
+            AND now()::timestamp <@ tsrange(
```

Comment:
I haven't looked at much else of the PR, but we're going to want to double check that this tsrange operation uses an index.

We also don't need a tsrange, right? If it's just syntactic sugar it might be worth avoiding so we can use `$queryRaw`.

Reply:
I found `tsrange` a lot easier to reason about but you were right, it wasn't using an index. Range queries like `<@` also require a special index.

As suggested, I changed it to use `$queryRaw` with `<` and `>` which can use a b-tree index in ccbde0d3.